tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -278,6 +278,99 @@ void toCameraInfo(const TangoCameraIntrinsics& camera_intrinsics,
     LOG(ERROR) << "Unknown camera ID: " << camera_intrinsics.camera_id;
   }
 }
+// Converts TangoCameraIntrinsics to Tango3DR_CameraCalibration.
+// @param camera_intrinsics, TangoCameraIntrinsics to convert.
+// @param t3dr_camera_intrinsics, the output Tango3DR_CameraCalibration.
+void toTango3DR_CameraCalibration(
+    const TangoCameraIntrinsics& camera_intrinsics,
+    Tango3DR_CameraCalibration* t3dr_camera_intrinsics) {
+  t3dr_camera_intrinsics->calibration_type =
+          static_cast<Tango3DR_TangoCalibrationType>(camera_intrinsics.calibration_type);
+  t3dr_camera_intrinsics->width = camera_intrinsics.width;
+  t3dr_camera_intrinsics->height = camera_intrinsics.height;
+  t3dr_camera_intrinsics->fx = camera_intrinsics.fx;
+  t3dr_camera_intrinsics->fy = camera_intrinsics.fy;
+  t3dr_camera_intrinsics->cx = camera_intrinsics.cx;
+  t3dr_camera_intrinsics->cy = camera_intrinsics.cy;
+  std::copy(std::begin(camera_intrinsics.distortion),
+            std::end(camera_intrinsics.distortion),
+            std::begin(t3dr_camera_intrinsics->distortion));
+}
+// Converts TangoPoseData to Tango3DR_Pose.
+// @param tango_pose_data, TangoPoseData to convert.
+// @param t3dr_pose, the output Tango3DR_Pose.
+void toTango3DR_Pose(const TangoPoseData& tango_pose_data, Tango3DR_Pose* t3dr_pose) {
+  std::copy(std::begin(tango_pose_data.translation),
+            std::end(tango_pose_data.translation),
+            std::begin(t3dr_pose->translation));
+  std::copy(std::begin(tango_pose_data.orientation),
+            std::end(tango_pose_data.orientation),
+            std::begin(t3dr_pose->orientation));
+}
+// Converts Tango3DR_Mesh to visualization_msgs::Marker (TRIANGLE_LIST).
+// See ROS documentation:
+// http://docs.ros.org/api/visualization_msgs/html/msg/Marker.html
+// @param grid_index, index of the tango_mesh.
+// @param tango_mesh, Tango3DR_Mesh to convert.
+// @param mesh_marker, the output visualization_msgs::Marker.
+void toMeshMarker(const Tango3DR_GridIndex& grid_index,
```

Comment:
Suggestion for possible follow up: create a file with free conversion functions for all Tango <-> ROS conversions. This will greatly reduce the number of code lines in this file.